### PR TITLE
Add continue train test case.

### PR DESCRIPTION
**Pull request by ltxtech1999**
Add test to confirm that lightgbm-transform support continue train.

### CODE REVIEW

File: tests/python_package_test/test_transform.py [hunk 1]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
...
+
+def test_continue_training(params, trained_model_path):
+    train_data = lgb.Dataset(rank_ds.data, params={"parser_config_file": rank_ds.parser_config}, free_raw_data = False)
+    bst = lgb.train(params, train_data, valid_sets=[train_data])
+    bst.save_model(trained_model_path)
+    print("Option 1 current iter# %d" %bst.current_iteration())
+    print("saved model")
+    params['num_trees'] = 20
+    params['num_leaves'] = 63
+    gbm = lgb.train(params, train_data, valid_sets=[train_data], init_model=bst)
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:162]: use more clear variable name like `continue_train_bst`

File: tests/python_package_test/test_transform.py [hunk 2]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
     bst = lgb.train(binary_params, train_data, valid_sets=[train_data])
     pred = bst.predict(simple_ds.data)
     np.testing.assert_allclose(pred[:5], np.array([0.4894574, 0.43920928, 0.71112129, 0.43920928, 0.39602784]))
+
+def test_continue_training(params, trained_model_path):
+    train_data = lgb.Dataset(rank_ds.data, params={"parser_config_file": rank_ds.parser_config}, free_raw_data = False)
+    bst = lgb.train(params, train_data, valid_sets=[train_data])
+    bst.save_model(trained_model_path)
+    print("Option 1 current iter# %d" %bst.current_iteration())
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:158]: change to `assert cur iteration number`.

File: tests/python_package_test/test_transform.py [hunk 3]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
...
+    print("saved model")
+    params['num_trees'] = 20
+    params['num_leaves'] = 63
+    gbm = lgb.train(params, train_data, valid_sets=[train_data], init_model=bst)
+    gbm.save_model('refit_model.txt')
+    print("Option 2 current iter# %d" %gbm.current_iteration())
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:164]: change to assert cur iteration number.

File: tests/python_package_test/test_transform.py (hunk 1 above)
Comment (ltxtech1999) [reply tests/python_package_test/test_transform.py]: Finished

File: tests/python_package_test/test_transform.py (hunk 2 above)
Comment (ltxtech1999) [reply tests/python_package_test/test_transform.py]: Use assert to confirm now.

File: tests/python_package_test/test_transform.py (hunk 3 above)
Comment (ltxtech1999) [reply tests/python_package_test/test_transform.py]: Finished

File: tests/python_package_test/test_transform.py [hunk 4]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
     bst = lgb.train(binary_params, train_data, valid_sets=[train_data])
     pred = bst.predict(simple_ds.data)
     np.testing.assert_allclose(pred[:5], np.array([0.4894574, 0.43920928, 0.71112129, 0.43920928, 0.39602784]))
+
+def test_continue_training(params, trained_model_path):
+    train_data = lgb.Dataset(rank_ds.data, params={"parser_config_file": rank_ds.parser_config}, free_raw_data = False)
+    bst = lgb.train(params, train_data, valid_sets=[train_data])
+    bst.save_model(trained_model_path)
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:157]: looks no need to save in this test.

File: tests/python_package_test/test_transform.py [hunk 5]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
...
+    assert bst.current_iteration() == 10
+    print("saved model")
+    params['num_trees'] = 20
+    params['num_leaves'] = 63
+    continue_train_bst = lgb.train(params, train_data, valid_sets=[train_data], init_model=bst)
+    continue_train_bst.save_model('refit_model.txt')
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:163]: looks no need to save in this test.

File: tests/python_package_test/test_transform.py [hunk 6]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
...
+    assert continue_train_bst.current_iteration() == 30
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:164]: Add a comment here to explain why the test prove continue train works.

File: tests/python_package_test/test_transform.py [hunk 7]
```diff
@@ -150,3 +150,15 @@ def test_train_label_id_less_than_transformed_feature_num(binary_params):
     bst = lgb.train(binary_params, train_data, valid_sets=[train_data])
     pred = bst.predict(simple_ds.data)
     np.testing.assert_allclose(pred[:5], np.array([0.4894574, 0.43920928, 0.71112129, 0.43920928, 0.39602784]))
+
```
Comment (chjinche) [inline tests/python_package_test/test_transform.py:153]: double blank lines to follow pep8 format.

File: tests/python_package_test/test_transform.py (hunk 6 above)
Comment (ltxtech1999) [reply tests/python_package_test/test_transform.py:164]: Finished

**Review by chjinche — APPROVED** [review]

**Pull request merged**